### DISCUSSION TITLE
PHP - Expenses Bundle Invalid - Docker Plugin Upgrade

### DISCUSSION THREAD

**Discussion by DustyArmstrong** [kimai/kimai#4153] [On-Premise: Installation & Updates] [answered]
### Describe the issue

Just upgraded my docker container to v2 of Kimai, and my Expenses Plugin is now broken. I recall receiving an email about upgrading my plugin so this wasn't an unknown.

`NOTICE: PHP message: PHP Fatal error:  Uncaught Exception: Bundle "ExpensesBundle" defines an invalid Kimai minimum version in extra.kimai.require. Please provide an integer as in Constants::VERSION_ID. in /opt/kimai/src/Plugin/PluginMetadata.php:59`

Essentially I am just requesting information on how to upgrade the docker container (via `docker-compose pull`) and the Expenses plugin without losing any data as I'm unable to reliably upgrade to v2 at the moment.

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.0.26

### How do you run Kimai?

Docker

### Which PHP version are you using?

8.1

### Logfile

`No such file or directory.`

**Comment by kevinpapst** [comment, marked as answer]
I do not support docker, but I guess the information works for containers as well:

See https://www.kimai.org/documentation/updates.html#updating-kimai
You have to remove the plugin directory first, flush the cache, then run the upgrade process.
Once 2.0 is running, install the plugin as described )which will update the plugin database automatically as well).
